assets/assets.go

```diff
@@ -29,6 +36,35 @@ var initDocPaths = []string{
 	filepath.Join("init-doc", "ping"),
 }
 
+func init() {
+	sha := sha256.New()
+	fs.WalkDir(dir, ".", func(path string, d fs.DirEntry, err error) error {
+		if d.IsDir() {
+			return nil
+		}
+
+		file, err := dir.ReadFile(path)
+		if err != nil {
+			panic(err)
+		}
+		_, err = sha.Write(file)
+		if err != nil {
+			panic(err)
+		}
+		return nil
+	})
+
+	hexSum := sha.Sum(nil)
+	AssetHash = fmt.Sprintf("%x", hexSum)
```

Comment:
Sum suggestion:
```suggestion
	sum := xxhash.New32()
	err := fs.WalkDir(dir, ".", func(path string, d fs.DirEntry, err error) error {
		if d.IsDir() {
			return nil
		}
		
		file, err := dir.Open(path)
		if err != nil {
			return err
		}
		defer file.Close()
		_, err = io.Copy(sum, file)
		return err
	})
	if err != nil {
		panic("error creating asset ETag sum: " + err.Error())
	}
	
	AssertHash = strconv.FormatUint(uint64(sum.Sum32()), 32)
```

Reply:
I like this proposal except for 1 thing: I dont think we should know about the `ETag` usage. At the end of the day, that is how this package is used by its consumer - but that could change over time and we wouldn't want to have the `Etag` through off.